lib/internal/process/methods.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const {
+  ERR_INVALID_ARG_TYPE,
+  ERR_INVALID_ARG_VALUE,
+  ERR_UNKNOWN_CREDENTIAL
+} = require('internal/errors').codes;
+const {
+  validateUint32
+} = require('internal/validators');
+
+function setupProcessMethods() {
+  // Non-POSIX platforms like Windows don't have certain methods.
+  if (process.setgid !== undefined) {
+    setupPosixMethods();
+  }
+
+  const {
+    chdir: _chdir,
+    umask: _umask,
+  } = process;
+
+  process.chdir = chdir;
+  process.umask = umask;
+
+  function chdir(directory) {
+    if (typeof directory !== 'string') {
+      throw new ERR_INVALID_ARG_TYPE('directory', 'string', directory);
+    }
+    return _chdir(directory);
+  }
+
+  const octalReg = /^[0-7]+$/;
+  function umask(mask) {
+    if (typeof mask === 'undefined') {
+      return _umask(mask);
+    }
+
+    if (typeof mask === 'number') {
+      validateUint32(mask, 'mask');
+      return _umask(mask);
+    }
+
+    if (typeof mask === 'string') {
+      if (!octalReg.test(mask)) {
+        throw new ERR_INVALID_ARG_VALUE('mask', mask,
+                                        'must be an octal string');
+      }
+      const octal = Number.parseInt(mask, 8);
+      validateUint32(octal, 'mask');
```

Comment:
This is actually only testing for `<= 2 ** 32`. All other checks are already done due to the RegExp above.
I personally would either remove the extra check or make the whole function stricter to only accept valid entries.
The reason `validateUint32` is used for numbers is only about making sure no negative numbers got passed in (as far as I know).

Reply:
Do you know what is the largest allowed value?

Reply:
What happens in the system if the value is larger than `0o777` ?
For example if you pass `0o12345` ? is it truncated to `0o345` ?

Reply:
Suggestion: add a TODO to investigate and land the PR. In `fs` there are a couple of validations that need improvements, especially having more consistent checks. I am also working on some of those.